source/modals/bountypostmodal.js

```diff
@@ -0,0 +1,112 @@
+const { GuildScheduledEventEntityType } = require('discord.js');
+const { database } = require('../../database');
+const { InteractionWrapper } = require('../classes');
+const { YEAR_IN_MS } = require('../constants');
+const { getRankUpdates } = require('../helpers');
+
+const customId = "bountypostmodal";
+module.exports = new InteractionWrapper(customId, 3000,
+	/** Serialize data into a bounty, then announce with showcase embed */
+	async (interaction, [slotNumber, isEvergreen]) => {
+		const title = interaction.fields.getTextInputValue("title");
+		const description = interaction.fields.getTextInputValue("description");
+		const imageURL = interaction.fields.getTextInputValue("imageURL");
+		const startTimestamp = parseInt(interaction.fields.getTextInputValue("startTimestamp"));
+		const endTimestamp = parseInt(interaction.fields.getTextInputValue("endTimestamp"));
+		const shouldMakeEvent = startTimestamp && endTimestamp;
+
+		const rawBounty = {
+			userId: interaction.user.id,
+			guildId: interaction.guildId,
+			slotNumber: parseInt(slotNumber),
+			isEvergreen: isEvergreen === "true",
+			title,
+			description
+		};
+		const errors = [];
+
+		if (imageURL) {
+			try {
+				new URL(imageURL);
+				rawBounty.attachmentURL = imageURL;
+			} catch (error) {
+				errors.push(error.message);
+			}
+		}
+
+		if (shouldMakeEvent) {
+			if (!startTimestamp) {
+				errors.push("Start timestamp must be an integer.");
+			} else if (!endTimestamp) {
+				errors.push("End timestamp must be an integer.");
+			} else {
+				if (startTimestamp > endTimestamp) {
+					errors.push("End timestamp was before start timestamp.");
+				}
+
+				const nowTimestamp = Date.now() / 1000;
+				if (nowTimestamp >= startTimestamp) {
+					errors.push("Start timestamp must be in the future.");
+				}
+
+				if (nowTimestamp >= endTimestamp) {
+					errors.push("End timestamp must be in the future.");
+				}
+
+				if (startTimestamp >= nowTimestamp + (5 * YEAR_IN_MS)) {
+					errors.push("Start timestamp cannot be 5 years in the future or further.");
+				}
+
+				if (endTimestamp >= nowTimestamp + (5 * YEAR_IN_MS)) {
+					errors.push("End timestamp cannot be 5 years in the future or further.");
+				}
+			}
+		}
+
+		if (errors.length > 0) {
+			interaction.reply({ content: `The following errors were encountered while posting your bounty **${title}**:\n• ${errors.join("\n• ")}`, ephemeral: true });
+		} else {
```

Comment:
To avoid needing an explicit `else` block, it would be better to just return in the `if` block here, since it also ends the validation steps.

Reply:
Fixed in https://github.com/Imaginary-Horizons-Productions/BountyBot/pull/23/commits/1539b761523841a8428a3e214bc2ef5bbcf964cb